File: src/clase-05/practica/typescript/activity4_facade_starter/subsystems.ts
export class Database {
  connect(): string { return "DB: conexión establecida"; }
  query(q: string): string { return `DB: ejecutando '${q}'`; }
  close(): string { return "DB: conexión cerrada"; }
}

export class Auth {
  init(): string { return "Auth: inicializado"; }
  login(user: string): string { return `Auth: login de ${user} ok`; }
  shutdown(): string { return "Auth: apagado"; }
}

export class Logger {
  start(): string { return "Logger: activo"; }
  log(msg: string): string { return `Logger: ${msg}`; }
  stop(): string { return "Logger: detenido"; }
}

File: src/clase-05/practica/python/activity4_facade_starter/subsystems.py
class Database:
    def connect(self) -> str:
        return "DB: conexión establecida"
    def query(self, q: str) -> str:
        return f"DB: ejecutando '{q}'"
    def close(self) -> str:
        return "DB: conexión cerrada"

class Auth:
    def init(self) -> str:
        return "Auth: inicializado"
    def login(self, user: str) -> str:
        return f"Auth: login de {user} ok"
    def shutdown(self) -> str:
        return "Auth: apagado"

class Logger:
    def start(self) -> str:
        return "Logger: activo"
    def log(self, msg: str) -> str:
        return f"Logger: {msg}"
    def stop(self) -> str:
        return "Logger: detenido"
